File: rocket-chat-core/src/main/java/chat/rocket/core/repositories/UserRepository.java
package chat.rocket.core.repositories;

import com.hadisatrio.optional.Optional;
import io.reactivex.Flowable;

import java.util.List;
import chat.rocket.core.SortDirection;
import chat.rocket.core.models.User;

public interface UserRepository {

  Flowable<Optional<User>> getCurrent();

  Flowable<List<User>> getSortedLikeName(String name, SortDirection direction, int limit);
}

File: rocket-chat-core/src/main/java/chat/rocket/core/repositories/UserRepository.kt
package chat.rocket.core.repositories

import chat.rocket.core.models.User
import com.hadisatrio.optional.Optional
import io.reactivex.Flowable

interface UserRepository {

    fun getAll(): Flowable<List<User>>

    fun getCurrent(): Flowable<Optional<User>>

    fun getByUsername(username: String): Flowable<Optional<User>>

    fun getSortedLikeName(name: String, limit: Int): Flowable<List<User>>
}